server-common/src/main/java/org/apache/gravitino/server/authorization/GravitinoAuthorizer.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.gravitino.server.authorization;
+
+import java.io.Closeable;
+import org.apache.gravitino.MetadataObject;
+import org.apache.gravitino.authorization.Privilege;
+
+/** Used for metadata authorization. */
+public interface GravitinoAuthorizer extends Closeable {
+
+  /**
+   * After instantiating the GravitinoAuthorizer, execute the initialize method to perform a series
+   * of initialization operations, such as loading privilege policies and so on.
+   */
+  void initialize();
+
+  /**
+   * Perform authorization and return the authorization result.
+   *
+   * @param userId the user id in use_meta table
+   * @param metadataType for example, CATALOG, SCHEMA,TABLE, etc.
+   * @param metadataId the metadata id.
+   * @param privilege for example, CREATE_CATALOG, CREATE_TABLE, etc.
+   * @return authorization result.
+   */
+  boolean authorize(
+      long userId, MetadataObject.Type metadataType, long metadataId, Privilege.Name privilege);
+}
```

Comment:
Sometimes multiple privileges can allow one action. Such as `load table` can use the privilege`select table` or `modify table`.

Reply:
OGNL is used to execute complex authorization logic. For example, if having either the `load table` or `select_table` permission grants access to the `load table` permission, OGNL can be used to call the `authorize` method of `GravitinoAuthorizer` multiple times and perform Boolean operations to obtain the final authorization result. (Refer to #6829)

Reply:
The **GravitinoAuthorizer** is only used for verifying a single privilege. Complex authorization expressions are executed through the **AuthorizationExpressionExecutor**, which will invoke the **GravitinoAuthorizer** using OGNL expressions and perform Boolean operations to obtain the final verification result.

Reply:
What's the full name of OGNL?

Reply:
[OGNL](https://github.com/orphan-oss/ognl) stands for Object-Graph Navigation Language; it is an expression language for getting and setting properties of Java objects, plus other extras such as list projection and selection and lambda expressions. You use the same expression for both getting and setting the value of a property.



I plan to use it to convert the expressions defined in **ExpressionsAuthorizeApi** into calls to **GravitinoAuthorizer.authorize** to implement complex authorization logic.

Reply:
OK, let us add the comment.